nucypher/crypto/utils.py

```diff
@@ -41,45 +44,93 @@ def construct_policy_id(label: bytes, stamp: bytes) -> bytes:
-        if not len(signature) == expected_signature_size:
-            raise ValueError(f"When passing a v value, "
-                             f"the signature size should be {expected_signature_size} B.")
+    if v_value_to_try in (0, 1, 27, 28):
         if v_value_to_try >= 27:
             v_value_to_try -= 27
         signature = signature + v_value_to_try.to_bytes(1, 'big')
     else:
         raise ValueError("Wrong v value. It should be 0, 1, 27 or 28.")
 
+    kwargs = dict(hasher=None) if is_prehashed else {}
     pubkey = PublicKey.from_signature_and_message(serialized_sig=signature,
-                                                  message=prehashed_message,
-                                                  hasher=None)
+                                                  message=message,
+                                                  **kwargs)
     return pubkey.format(compressed=True)
+
+
+def get_signature_recovery_value(message: bytes,
+                                 signature: Union[bytes, Signature],
+                                 public_key: Union[bytes, UmbralPublicKey],
+                                 is_prehashed: bool = False) -> bytes:
+    """
+    Obtains the recovery value of a standard ECDSA signature.
+
+    :param message: Signed message
+    :param signature: The signature from which the pubkey is recovered
+    :param public_key: The public key for verifying the signature
+    :param is_prehashed: True if the message is already pre-hashed. Default is False, and message will be hashed with SHA256
+    :return: The compressed byte-serialized representation of the recovered public key
+    """
+
+    signature = bytes(signature)
+    ecdsa_signature_size = Signature.expected_bytes_length()
+    if len(signature) != ecdsa_signature_size:
+        raise ValueError(f"The signature size should be {ecdsa_signature_size} B.")
+
+    kwargs = dict(hasher=None) if is_prehashed else {}
+    for v in (0, 1):
+        v_byte = bytes([v])
+        recovered_pubkey = PublicKey.from_signature_and_message(serialized_sig=signature + v_byte,
+                                                                message=message,
+                                                                **kwargs)
+        if bytes(public_key) == recovered_pubkey.format(compressed=True):
+            return v_byte
+    else:
+        raise ValueError("Signature recovery failed. "
+                         "Either the message, the signature or the public key is not correct")
+
+
+def get_coordinates_as_bytes(point: Union[Point, UmbralPublicKey, SignatureStamp],
+                             x_coord=True,
+                             y_coord=True) -> bytes:
+    if isinstance(point, SignatureStamp):
+        point = point.as_umbral_pubkey()
+
+    coordinates_as_bytes = point.to_bytes(is_compressed=False)[1:]
+    middle = len(coordinates_as_bytes)//2
+    if x_coord and y_coord:
+        return coordinates_as_bytes
```

Comment:
Would it be better to return this as a tuple? Seems like this would be a better function that way, but this works too.

Reply:
Not really, since this function is used to get a single bytes sequence with the coords.